SEKOIA-IO/intake-formats
# Migrate Windows to Ingest

Current state of this migraton


This fields have different types in Ingest than in Logstash:

|field	| type logstash	| type ingest
|---|---|---|
|event.code	|int	|keyword
|action.properties.BytesTotal	|int	|keyword
|action.properties.Task	|string	|int
|action.properties.LastASSecurityIntelligenceAge	|int	|string
|action.properties.LastAVSecurityIntelligenceAge	|int	|string
|action.properties.LastFullScanAge	|int	|string
|action.properties.LastQuickScanAge	|int	|string
|action.properties.DestinationPort	|int	|string

Improvements:
- The `action.properties.Keywords` value was broken in Logstash.
- The NBSP from the ContextInfo field was removed https://github.com/SEKOIA-IO/intake-formats/blob/06c0885a251a78f14646555c3d4770c10e64d1d4/Windows/windows/ingest/parser.yml#L322
- `process.command_line` automatically composed by the finalizer. We have a better result than the previous which was a concatenation (working_directory + command_line):

1. example of diff
```diff
- 'name': 'ruby.exe',
-  'working_directory': 'c:\\program files\\puppet labs\\puppet\\puppet\\bin\\',
- 'command_line': 'c:\\program files\\puppet labs\\puppet\\puppet\\bin\\c:/program files/puppet labs/puppet/puppet/bin/ruby.exe c:/program files/puppet labs/puppet/puppet/bin/puppet config --section agent --log_level notice print runinterval ',
-  'executable': 'c:\\program files\\puppet labs\\puppet\\puppet\\bin\\ruby.exe'},
+ 'command_line': 'c:/program files/puppet labs/puppet/puppet/bin/ruby.exe c:/program files/puppet labs/puppet/puppet/bin/puppet config --section agent --log_level notice print runinterval ',
+ 'executable': 'c:\\program files\\puppet labs\\puppet\\puppet\\bin\\ruby.exe',
+ 'name': 'ruby.exe',
+ 'working_directory': 'c:\\program files\\puppet labs\\puppet\\puppet\\bin\\'
  ```
2. example of diff

```diff
- 'name': 'powershell.exe',
- 'working_directory': 'c:\\windows\\system32\\windowspowershell\\v1.0\\',
- 'command_line': 'c:\\windows\\system32\\windowspowershell\\v1.0\\powershell.exe -file c:\\exploitation\\scripts\\nagios\\get-timelaps.ps1',
- 'executable': 'c:\\windows\\system32\\windowspowershell\\v1.0\\powershell.exe'},
+ 'command_line': 'powershell.exe -file c:\\exploitation\\scripts\\nagios\\get-timelaps.ps1',
+ 'executable': 'c:\\windows\\system32\\windowspowershell\\v1.0\\powershell.exe',
+ 'name': 'powershell.exe',
+ 'working_directory': 'c:\\windows\\system32\\windowspowershell\\v1.0\\'
  ```

Further improvements to be added latter:
- remove calls to `| lower`
- remove fields value containing `-` or `N/A`

Unfixable test:
- the working_dir test has a `input[log']['hostname']` https://github.com/SEKOIA-IO/intake-formats/pull/585/files#diff-0eb3aee69f6c5fa0cb040e25545d736ae483ad922b2887441352f97f2a4883b7R7-R9. We did not know what it was intended for.
The tests are the unmodified ones from Logstash.
